searx/engines/baidu.py

```diff
@@ -0,0 +1,64 @@
+# SPDX-License-Identifier: AGPL-3.0-or-later
+"""Baidu
+
+.. Website: https://www.baidu.com
+"""
+
+from urllib.parse import urlencode
+from datetime import datetime
+
+from searx.exceptions import SearxEngineAPIException
+
+about = {
+    "website": "https://www.baidu.com",
+    "wikidata_id": "Q14772",
+    "official_api_documentation": None,
+    "use_official_api": False,
+    "require_api_key": False,
+    "results": "JSON",
+}
+
+paging = True
+results_per_page = 10
+categories = ["general"]
+
+base_url = "https://www.baidu.com/s"
```

Comment:
```suggestion
base_url = "http://www.baidu.com/s"
```
This is a http api.
![image](https://github.com/user-attachments/assets/93929ba4-f6ce-414d-8519-bd1f082ae0f0)

Reply:
This seems to be related to UA. If it's the browser's UA, it can return JSON. In addition, `httpx` defaults to rejecting access to http, so it's kept as https for now.